src/components/Project/test/ProjectExecutions.test.tsx

```diff
@@ -20,59 +20,116 @@ import {
     let failedTaskFixture: ReturnType<typeof oneFailedTaskWorkflow.generate>;
-    let executions: Execution[];
+    let executions1: Execution[];
+    let executions2: Execution[];
     let scope: DomainIdentifierScope;
     let queryClient: QueryClient;
+    let mockGetUserProfile: jest.Mock<ReturnType<typeof getUserProfile>>;
+
+    const sampleUserProfile: UserProfile = {
+        sub: 'sub'
+    } as UserProfile;
+
+    const defaultQueryParams1 = {
+        [sortQueryKeys.direction]: SortDirection[SortDirection.DESCENDING],
+        [sortQueryKeys.key]: executionSortFields.createdAt
+    };
+
+    const defaultQueryParams2 = {
+        filters: 'eq(user,sub)',
+        [sortQueryKeys.direction]: SortDirection[SortDirection.DESCENDING],
+        [sortQueryKeys.key]: executionSortFields.createdAt
+    };
 
     beforeEach(() => {
+        mockGetUserProfile = jest.fn().mockResolvedValue(null);
         queryClient = createTestQueryClient();
         basicPythonFixture = basicPythonWorkflow.generate();
         failedTaskFixture = oneFailedTaskWorkflow.generate();
         insertFixture(mockServer, basicPythonFixture);
         insertFixture(mockServer, failedTaskFixture);
-        executions = [
+        executions1 = [
             basicPythonFixture.workflowExecutions.top.data,
             failedTaskFixture.workflowExecutions.top.data
         ];
-        const { domain, project } = executions[0].id;
+        executions2 = [];
+        const { domain, project } = executions1[0].id;
         scope = { domain, project };
         mockServer.insertWorkflowExecutionList(
             scope,
-            executions,
-            defaultQueryParams
+            executions1,
+            defaultQueryParams1
+        );
+        mockServer.insertWorkflowExecutionList(
+            scope,
+            executions2,
+            defaultQueryParams2
         );
     });
 
     const renderView = () =>
         render(
             <QueryClientProvider client={queryClient}>
-                <ProjectExecutions
-                    projectId={scope.project}
-                    domainId={scope.domain}
-                />
+                <APIContext.Provider
+                    value={mockAPIContextValue({
+                        getUserProfile: mockGetUserProfile
+                    })}
+                >
+                    <ProjectExecutions
+                        projectId={scope.project}
+                        domainId={scope.domain}
+                    />
+                </APIContext.Provider>
             </QueryClientProvider>,
             { wrapper: MemoryRouter }
         );
 
-    it('displays executions after successful load', async () => {
-        const { getByText } = renderView();
-        await waitFor(() => expect(getByText(executions[0].id.name)));
+    it('should not display checkbox if user does not login', async () => {
+        const { queryByTestId } = renderView();
+        await waitFor(() => {});
```

Comment:
Prefer not to use empty `waitFor` functions. You can put your assertions into the body of the function and the test will wait until they pass.
Also, you can check for the checkbox by label instead of using a test-id.
```ts
await waitFor(() => {
    expect(mockGetUserProfile).toHaveBeenCalled();
    expect(queryByLabel('Only my executions')).toBeNull();
});
```